### ISSUE TITLE
undefined method '[]' for nil with (incomplete) case statement

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14100]
## Steps to reproduce the problem

```bash
$ rubocop -V
1.75.2 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.44.1, analyzing as Ruby 3.4, running on ruby 3.4.3) [arm64-darwin24]

$ cat > foo.yml
AllCops:
  TargetRubyVersion: 3.4.3
^D
$ cat > foo.rb
```
```ruby
2.times do
  case 1
  when
  end
end
```
```bash
^D
$ rubocop -c ~/foo.yml foo.rb
undefined method '[]' for nil
/opt/homebrew/lib/ruby/gems/3.4.0/gems/prism-1.4.0/lib/prism/translation/parser/lexer.rb:310:in 'Prism::Translation::Parser::Lexer#to_a'
# FLOOD
```

**Comment by Earlopain** [comment]
Thanks for the report! I openend https://github.com/ruby/prism/pull/3551, which should fix this (and other similar issues with invalid syntax)

**Comment by Earlopain** [comment]
Will be fixed in the next prism release.

**Comment by Earlopain** [comment]
prism 1.5.0 got released and doesn't raise anymore, so this can be closed now.

**Comment by forthrin** [comment]
Confirmed.
